### ISSUE TITLE
Go SDK version v1.0.0-testplsignore.0 invalid

### ISSUE THREAD

**Issue by dinesh-auti** [pulumi/pulumi-awsx#871]
### What happened?

While trying to set up a basic VPC using AWSX in golang I get a 403 on a plugin install.

```
➜ vpcx pulumi plugin install resource awsx v1.0.0-testplsignore.0 
[resource plugin awsx-1.0.0-testplsignore.0] installing
error: [resource plugin awsx-1.0.0-testplsignore.0] downloading from : 403 HTTP error fetching plugin from https://get.pulumi.com/releases/plugins/pulumi-resource-awsx-v1.0.0-testplsignore.0-darwin-amd64.tar.gz
```

### Steps to reproduce

I have the below code to setup a basic VPC using awsx. 

```
package main

import (
	"log"
	ec2x "github.com/pulumi/pulumi-awsx/sdk/go/awsx/ec2"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

func main() {

	cidr := "10.0.0.0/16"

	pulumi.Run(func(ctx *pulumi.Context) error {
		_, err := ec2x.NewVpc(ctx, "test-vpc", &ec2x.VpcArgs{
			CidrBlock: &cidr,
		})
		if err != nil {
			log.Fatalln(err)
		}

		return nil
	})
}
```

Once I run `pulumi up`, I get the below error 
```
  error: no resource plugin 'pulumi-resource-awsx' found in the workspace at version v1.0.0-testplsignore.0 or on your $PATH, install the plugin using `pulumi plugin install resource awsx v1.0.0-testplsignore.0`
```

I try installing the said plugin but it fails too 
```
➜ vpcx pulumi plugin install resource awsx v1.0.0-testplsignore.0 
[resource plugin awsx-1.0.0-testplsignore.0] installing
error: [resource plugin awsx-1.0.0-testplsignore.0] downloading from : 403 HTTP error fetching plugin from https://get.pulumi.com/releases/plugins/pulumi-resource-awsx-v1.0.0-testplsignore.0-darwin-amd64.tar.gz
```

### Expected Behavior

Expected behavior is to show a preview after `pulumi up`

### Actual Behavior

I get the below error on `pulumi up`

```
error: no resource plugin 'pulumi-resource-awsx' found in the workspace at version v1.0.0-testplsignore.0 or on your $PATH, install the plugin using `pulumi plugin install resource awsx v1.0.0-testplsignore.0`
```

### Versions used

```
➜ ~ pulumi about
CLI          
Version      3.34.0
Go Version   go1.17.10
Go Compiler  gc

Host     
OS       darwin
Version  12.2.1
Arch     x86_64

Backend        
Name           ******
URL            s3://*******?profile=dev
User           ****
Organizations  
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by danielrbradley** [comment]
Hi @dinesh-auti please could you tell me which version of the awsx go-SDK you're using (what's in your go.mod)?

The version `v1.0.0-testplsignore.0` isn't a valid published version, it should look something like `v1.0.0-beta.8`. You might be able to work around this issue by running `pulumi plugin install resource awsx 1.0.0-beta.8`.

**Comment by dinesh-auti** [comment]
I see this.

```
go 1.17

require (
	github.com/pulumi/pulumi-awsx/sdk v1.0.0-testplsignore.0
	github.com/pulumi/pulumi/sdk/v3 v3.34.1
)
```

Also, the latest version is v1.0.0-testplsignore.0 in the package doc - https://pkg.go.dev/github.com/pulumi/pulumi-awsx/sdk/go/awsx

<img width="501" alt="Screen Shot 2022-06-21 at 8 39 27 AM" src="https://user-images.githubusercontent.com/54119178/174801170-91d7deb5-1cd9-4ebc-99f0-ef715897f9cf.png">

**Comment by danielrbradley** [comment]
Ah this makes sense. We'll see if we can pull that incorrect tag. 

The syntax you're using the pull the specific version is incorrect - you need an `@` between the package and the version:

```
go get github.com/pulumi/pulumi-awsx/sdk@v1.0.0-beta.8
```

**Comment by dinesh-auti** [comment]
Ahh thanks! That works!

**Comment by DavidJFelix** [comment]
Looks like the tag is still published and go considers it "more recent" than latest beta tags.

**Comment by dinesh-auti** [comment]
I can confirm it is the latest.

**Comment by DavidJFelix** [comment]
![image](https://user-images.githubusercontent.com/263432/192303421-3e0e86e4-5406-401e-b59c-ce5b063bddfa.png)

From here:
https://pkg.go.dev/github.com/pulumi/pulumi-awsx/sdk/go/awsx/awsx?tab=versions

**Comment by danielrbradley** [comment]
Unfortuately, this seems to be a limitation in the go module cache. We've revoked the problematic version with a new beta release, but it appears that because "test" is alphnumerically sorted after "beta", any new beta version can't revoke a "test" version.

This is therefore only able to be resolved by releasing a new non-beta version. This is on the roadmap to address in the near future though.

**Comment by DavidJFelix** [comment]
`v1.0.1-beta.11` would be sorted ahead of `v1.0.0-testplsignor.0` without releasing any non-pre-release version. This is a little bit of a weird way to operate because:

* It may imply to people that v1.0.0 exists, which it does not and would not
* It may "soften the impact" of a "v1.0.0 release!"
* It means that your first non pre-release needs to be the same version as the beta, so: `v1.0.1` which may feel weird

Human considerations aside, this is the point of patch versions -- to add logistical fixes that don't have downstream impact. 

IDK. Either way I'm manually installing it until release, just some ideas for consideration.

**Comment by danielrbradley** [comment]
This should now be resolved since the release of the v1.0.0 Go SDK.

**Comment by DavidJFelix** [comment]
Confirmed
